src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,169 @@
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the file used to store existing task list data.
+ */
+public class Storage {
+    private static final DateTimeFormatter IN_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+    private String filePath;
+
+    /**
+     * Creates an instance of Storage.
+     *
+     * @param filePath filepath of the local file to store data
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Load the tasks data from the file if exists.
+     *
+     * @return ArrayList of the task read from file
+     * @throws DukeException exception for error while reading
+     */
+    public ArrayList<Task> load() throws DukeException {
+        File file = new File(Paths.get(filePath).toUri());
+        if (file.exists()) {
+            return readFile(file);
+        } else {
+            throw new DukeException("Reading Error");
+        }
+    }
+
+    /**
+     * Save single line of data at the end of the data file.
+     *
+     * @param task task to save to the file
+     * @throws DukeException exception for error while writing data to the file
+     */
+    public void saveSingle(Task task) throws DukeException {
+        saveCheck();
+        try {
+            FileWriter fw = new FileWriter(Paths.get(filePath).toString(), true);
+            fw.write(task.saveString() + System.lineSeparator());
+            fw.close();
+        } catch (IOException ex) {
+            throw new DukeException("Save single error.");
+        }
+
+    }
+
+    /**
+     * Save list of tasks to the data file.
+     *
+     * @param tasks list of task to save to the file
+     * @throws DukeException exception for error while writing data to the file
+     */
+    public void saveAll(TaskList tasks) throws DukeException {
+        saveCheck();
+        try {
+            FileWriter fw = new FileWriter(Paths.get(filePath).toString());
+            for (Task task : tasks.getList()) {
+                fw.write(task.saveString() + System.lineSeparator());
+            }
+            fw.close();
+        } catch (IOException ex) {
+            throw new DukeException("Save all error.");
+        }
+    }
+
+    /**
+     * Check if the file and directory exist before saving. Create the file and directory if do not exist.
+     *
+     * @throws DukeException exception for error while creating the file or directory
+     */
+    private void saveCheck() throws DukeException {
+        File file = new File(Paths.get(filePath).toUri());
+        boolean success;
+        if (!file.exists()) {
+            File dir = new File(file.getParentFile().getAbsolutePath());
+            if (!dir.exists()) {
+                success = dir.mkdir();
+                if (!success) {
+                    throw new DukeException("Create file directory error");
+                }
+            }
```

Comment:
the nested if-else here is quite jarring. consider wrapping this into a try-catch block where u catch an IOException instead.

Reply:
Thanks, I definitely will fix this nested if-else statement.